cli/export/export.go

```diff
@@ -0,0 +1,76 @@
+/*
+Tool to export annotation results from the internal DB to an output DB.
+
+Usage: export <path-to-origin.db> <path-to-destination.db>
+*/
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+
+	"github.com/jessevdk/go-flags"
+)
+
+const desc = `Exports annotation results from the internal input database to a new output
+database.
+The destination file must not exist.`
+
+var opts struct {
+	Args struct {
+		Input  string `description:"SQLite database filepath"`
+		Output string `description:"SQLite database filepath"`
+	} `positional-args:"yes" required:"yes"`
+}
+
+func main() {
+	parser := flags.NewParser(&opts, flags.Default)
+	parser.LongDescription = desc
+
+	if _, err := parser.Parse(); err != nil {
+		if err, ok := err.(*flags.Error); ok {
+			if err.Type == flags.ErrHelp {
+				os.Exit(0)
+			}
+
+			fmt.Println()
+			parser.WriteHelp(os.Stdout)
+		}
+
+		os.Exit(1)
+	}
+
+	if _, err := os.Stat(opts.Args.Input); os.IsNotExist(err) {
+		log.Fatalf("File %q does not exist", opts.Args.Input)
+	}
+
+	// TODO: until we have more information about the output schema needed,
+	// this tool does a simple copy&paste of the internal DB. The annotation
+	// results are stored in the assignments table.
+
+	if err := copy(opts.Args.Input, opts.Args.Output); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func copy(source, destination string) error {
+	src, err := os.Open(source)
+	if err != nil {
+		return err
+	}
+	defer src.Close()
+
+	dst, err := os.OpenFile(destination, os.O_RDWR|os.O_CREATE|os.O_TRUNC|os.O_EXCL, 0666)
+	if err != nil {
+		return err
+	}
+	defer dst.Close()
+
+	if _, err := io.Copy(dst, src); err != nil {
```

Comment:
The `if` is not necesary, you can also do:
```go
_, err := io.Copy(dst, src)
return err
```

Reply:
Thanks. To be honest, somehow my version feels cleaner to me. Separating the error handling and the last return with no error makes it easier to drop new lines by someone later. Otherwise the new author may forget the `if err!=nil return err`; and the new diff will show modifications to io.Copy when that is not really part of the new functionality.

Having said that, I'm not against this change. Is this considered the idiomatic way, or is it more a team convention?

Reply:
the standard library is the best place for inspiration:
https://github.com/golang/go/blob/master/src/net/http/response.go#L321
https://github.com/golang/go/blob/master/src/time/time.go#L1232

both versions are valid. I'm ok with extra checking for an error and returning nil otherwise. But I know some people don't like it.

Reply:
Removing the `if` doesn't mean that the error is not handled (I feel it's even more explicit that we're returning the error returned by the `io.Copy`). Once said that, I don't see the benefit of a redundant `if` when it's not necessary (considering that Go design does not focus in keeping the diffs as smaller as possible after every code modification)

Anyhow, since the stdlib is not consistent as Smaker pointed out, feel free to proceed on your own ;)

Reply:
I'm ok to always remove redundant `if`s if everybody else wants it. Should we vote?